lib/sorcery/controller.rb

```diff
@@ -63,6 +65,16 @@ def login(*credentials)
         end
       end
 
+      def login!(*credentials)
```

Comment:
We'll need to test this and see how it interacts when given a code block to yield to.

Reply:
I'm not sure what the behavior should be when this method is called with a block. What do you suggest?

Reply:
Essentially, sanity checks that the method performs as expected when passing a block.

This wiki page has an example of using the block for raising different error messages during login: https://github.com/Sorcery/sorcery/wiki/Distinguish-login-failure-reasons

When given a code block, we can test that it succeeds when given valid credentials, fails when given invalid credentials, and raises your new error when given a blank string or nil value.

Reply:
@joshbuker I added specs for the block version of `#login!`. I have a few notes:
- I can't find tests for the block version of `#login`
- I'm not sure what the behavior should be when calling `#login!` with a block and with wrong credentials. I would expect it to always raise the `InvalidCredentials` exception, else I don't see the value in using `#login!`